packages/dashboard/src/components/Section/examples/EditSections.Customization.Example.tsx

```diff
@@ -0,0 +1,280 @@
+        saveButtonDisabled={this.state.saveButtonDisabled}
+        defaultSectionTitle={'untitled section'}
+        defaultNewSectionId={'blank_id'}
+        deleteSectionDialogContentProps={{
+          title: 'Remove this section?',
+          subText: 'All the cards in this section will be removed. You can add them again to a new section.'
+        }}
+        deleteSectionDialogModelProps={{
+          isBlocking: false
+        }}
+        deleteSectionDilaogRemoveButtonProps={{
+          text: 'Remove',
+          ariaLabel: 'Remove section from dashboard'
+        }}
+        deleteSectionDilaogCancelButtonProps={{
+          text: 'Cancel',
+          ariaLabel: 'Cancel removing'
+        }}
+        renameSectionButtonProps={{
+          iconProps: { iconName: 'Edit' },
+          ariaLabel: 'Rename this section'
+        }}
+        deleteSectionButtonProps={{
+          iconProps: { iconName: 'Delete' },
+          ariaLabel: 'Delete this section from dashboard'
+        }}
+        addButtonProps={{
+          iconProps: { iconName: 'CircleAddition' },
+          text: 'Add a new section',
+          ariaLabel: 'Add a few section into dashboard'
+        }}
+        cancelButtonProps={{
+          iconProps: { iconName: 'Cancel' },
+          text: 'Cancel',
+          ariaLabel: 'Cancel changes'
+        }}
+        saveButtonProps={{
+          iconProps: { iconName: 'CheckMark' },
+          text: 'Save',
+          ariaLabel: 'Save changes'
+        }}
+        rowHeight={40}
+        onAddSection={this._onAddSection}
+        onRenameSectionClick={this._onRenameSectionClick}
+        onUpdateSectionTitle={this._onUpdateSectionTitle}
+        onDeleteSection={this._onDeleteSection}
+        onCancel={this._onCancel}
+        styles={{
+          icon: {
+            color: 'red'
+          }
+        }}
+      />
+    );
+  }
+
+  private _sections(): ISection[] {
+    const keys = ['0', '1', '2'];
+    return [
+      {
+        id: keys[0],
+        title: 'Basics'
+      },
+      {
+        id: keys[1],
+        title: 'Info security'
+      },
+      {
+        id: keys[2],
+        title: 'Users'
+      }
+    ];
+  }
+
+  /**
+   * Example handler for onLayoutChange.
+   * In actual implementation, should compare the new layout and old layout to avoid unnecessary saving api calls
+   */
+  private _onLayoutChange = (currentLayout: Layout[], allLayouts: Layouts): void => {
+    const newDashboardLayout = this._generateDashboardLayoutFromRGLLayout(allLayouts);
+
+    this.setState({
+      ...this.state,
+      layout: newDashboardLayout,
+      saveButtonDisabled: false
+    });
+  };
+
+  /**
+   * Example handler for onAddSection, which add a section with a random generated ID.
+   * In actuall implementation, could be improved by using a GUID as the new section ID
+   */
+  private _onAddSection = (title: string): void => {
+    const randomId = this._generateRandomNewSectionId();
```

Comment:
@caradong is there a reason why we need to use a random section Id?

Reply:
This is an example. That is why I left a comment in the documentation. 'In actuall implementation, could be improved by using a GUID as the new section ID'. The team needs to decided what to do in the M365UX code depends on what naming convention we will use.

---
In reply to: [219571053](https://github.com/OfficeDev/office-ui-fabric-react/pull/6358#discussion_r219571053) [](ancestors = 219571053)